src/modules/user/user.service.ts

```diff
@@ -33,18 +38,53 @@ export default class UserService {
     if (existingUser) {
       throw new Error("Email already in use.");
     } else {
-      const salt = await bcrypt.genSalt(10);
-      const hashedPassword = await bcrypt.hash(userInput.password, salt);
-
+      const password = await this.authService.createPassword(userInput);
       // returns the newly created user object to the resolver
       const user = await prisma.user.create({
         data: {
           email: userInput.email,
-          passwordHash: hashedPassword,
+          passwordHash: password,
           isActive: true,
         },
       });
       return user;
     }
   }
+
+  async deactivateAccount(email: string): Promise<string> {
+    const user = await prisma.user.findUnique({ where: { email } });
+
+    if (!user) {
+      throw new Error("User not found");
+    }
+
+    if (!user.isActive) {
+      throw new Error("User is already inactive");
+    }
+
+    await prisma.user.update({
+      where: { email },
+      data: { isActive: false },
+    });
+
+    // Set a timer to delete the account after 30 days
+    setTimeout(async () => {
+      await prisma.user.delete({ where: { email } });
+    }, 30 * 24 * 60 * 60 * 1000);
+
+    return "Your account has been deactivated and will be deleted in 30 days";
+  }
+
+  async deleteAccount(id: number): Promise<boolean> {
+    try {
+      await prisma.user.delete({
+        where: {
+          id,
+        },
+      });
+      return true;
```

Comment:
Maybe this can make use of the `AccountMessage` type? And it returns a status 200 with a your account has been permanently deleted message, and a 500, error. 

Then also, whenever we create more functions that need to send a message with a status, we can reuse this type!